src/GafferImageUI/ImageView.cpp

```diff
@@ -393,12 +432,86 @@ bool ImageView::keyPress( const GafferUI::KeyEvent &event )
 				);
 				return true;
 			}
+			if(event.key == "Home")
+			{
+				
+				V2i viewport = viewportGadget()->getViewport();
+				V2i viewportCenter(viewport.x / 2, viewport.y / 2);
+				Box3f bound = m_imageGadget->bound();
+				V2i imageCenter((bound.max.x - bound.min.x) / 2, (bound.max.y - bound.min.y) / 2);
+				Box3f frame(
+					Imath::V3f(imageCenter.x - viewportCenter.x, imageCenter.y - viewportCenter.y, 0),
+					Imath::V3f(imageCenter.x + viewportCenter.x, imageCenter.y + viewportCenter.y, 0)
+				);
+				viewportGadget()->frame(frame);
+				return true;
+			}
+            if(event.key == "Plus")
+            {
+                float zoomLevel = computeZoomLevel();
+                float roundLevel;
+
+                if((zoomLevel + EPSILON) < 1.f)
+                {
+                    zoomLevel = std::min(100.f, 1 / zoomLevel);
+                    roundLevel = round_down_to_even(zoomLevel);
+                    
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * .5;
+                    }
+                    roundLevel = 1 / roundLevel;
+                }
+                else
+                {
+                    roundLevel = round_up_to_even(zoomLevel);
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * 2;
+                    }
+                }
+                zoomLevelPlug()->setValue(roundLevel) ;
+				
+                return true;
+
+            }
+            if(event.key == "Minus")
+            {
+                float zoomLevel = computeZoomLevel();
+                float roundLevel;
+                if((zoomLevel - EPSILON) < 1.f)
+                {
+                    zoomLevel = std::min(1000.0, round(1 / zoomLevel));
+                    roundLevel = round_up_to_even(zoomLevel);
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * 2;
+                    }
+                    roundLevel = 1 / roundLevel;
+                }
+                else
+                {
+                    roundLevel = round_down_to_even(round(zoomLevel));
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * .5;
```

Comment:
Forgive me, I don't recall our discussion at Image Engine in detail. What do the level 2, 3 etc refer to in terms of zoom factors?

It seems that if there are very specific zoom factors you want this curve to hit (and which we would snap to), then a constant factor might not be appropriate? But assuming there is some constant factor that does fit, then I think the increment/decrement can be implement like so :

```
level = math.log( currentZoom, zoomFactor )
level = round( level ) # snap
level += 1
newZoom = math.pow( zoomFactor, level )
```